cronjobs/create_pipeline_execution.py

```diff
@@ -0,0 +1,230 @@
+#!/bin/python3.9
+# -*- coding: utf-8 -*-
+
+# Algorithm to bring pipeline phases in an ideal order
+
+import argparse
+from graphlib import TopologicalSorter
+
+class Phase:
+    key = "" 
+    mode = ""
+    title_norm = ""
+    name = ""
+    command = ""
+    def __init__(self, _key, _mode, _title_norm, _name):
+        self.key = _key
+        self.mode = _mode
+        self.title_norm = _title_norm
+        self.name = _name
+
+# Dependency list has to be built dynamically later
+# Preprocessing
+# Check Record Integrity at the Beginning of the Pipeline
+# marc_check --do-not-abort-on-empty-subfields --do-not-abort-on-invalid-repeated-fields \
+#            --write-data=GesamtTiteldaten-post-phase"$PHASE"-"${date}".mrc GesamtTiteldaten-"${date}".mrc >> "${log}" 2>&1
+
+# convention in dict-keys: 
+#   leading "_" can only run after predecessor has finished
+#   trailing "_" successor must not run before job has finished
+#   leading and trailing "_" has to run independently
+
+phase_counter = 0
+
+phases = {
```

Comment:
If you have no keys, you should use a list [] rather than a dictionary {}

Reply:
or, you add the key outside the phase object, then you do not need the get_phase_by_key function anymore